tests/checkify_test.py

```diff
@@ -1218,5 +1218,42 @@ def f(x):
                                 "x needs to be positive"):
       f(-1.)
 
+class InstrumentTest(jtu.JaxTestCase):
+  def test_gather(self):
+    def f(x, i):
+      with instrument():
+        return x[i]
+
+    err, out = catch(f)(jnp.ones((2,)), 3)
+    self.assertIsNotNone(err.get())
+    self.assertStartsWith(err.get(), "out-of-bounds indexing")
+
+    def f(x, i):
+      return x[i]
+
+    err, out = catch(f)(jnp.ones((2,)), 3)
+    self.assertIsNone(err.get())
+
+  def test_no_discharge_error(self):
+    self.skipTest("this fails")
+
+    @jax.jit
+    def f(x, i):
+      with instrument():
+        return x[i]
+
+    with self.assertRaisesRegex(ValueError, "Uncaught error!"):
```

Comment:
I think we want to throw this error through effects. The instrumented checks don't call a `checkify` specific primitive (ie. no `check_p` is used), so not sure how to add that effect. Could call `check_p` there too for consistency?

Reply:
aha, through the primitives `abstract_eval` rule ofc!